cca.go

```diff
@@ -0,0 +1,122 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"errors"
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix"
+	"github.com/gonum/matrix/mat64"
+)
+
+// CanonicalCorrelations returns the `canonical' correlations between
+// the corresponding pairs of canonical variables -- described by the
+// left and right eigenvectors of the canonical correlation matrix,
+// pvecs and qvecs respectively. The canonical correlation matrix is
```

Comment:
I had a go at re-writing this, seperating into more sentences addressing your other comments and attempting to address this one.